### PR TITLE
Refactor font passing.

### PR DESCRIPTION
**Pull request by ZehMatt**
Pretty early thing, now would be the time to leave suggestions and ideas about what those fonts are and how they are used.

### CODE REVIEW

File: src/openrct2/drawing/Text.cpp [hunk 1]
```diff
@@ -141,12 +141,12 @@ static void DrawTextEllipsisedCompat(
     DrawText(dpi, x, y, &_legacyPaint, buffer);
 }
 
-void gfx_draw_string(rct_drawpixelinfo* dpi, const_utf8string buffer, uint8_t colour, int32_t x, int32_t y)
+void gfx_draw_string(rct_drawpixelinfo* dpi, const_utf8string buffer, uint8_t colour, int32_t x, int32_t y, TextFont font)
```
Comment (duncanspumpkin) [inline src/openrct2/drawing/Text.cpp]: Since the vast majority of callers use Font medium maybe that should be a default value.

Comment (ZehMatt) [reply src/openrct2/drawing/Text.cpp]: I also wonder if I should perhaps rename it to TextSettings/TextOptions and put the color in there as well

Comment (Gymnasiast) [reply src/openrct2/drawing/Text.cpp]: I agree with Duncan. It makes for more readable code IMO.

**Comment by Gymnasiast** [comment]
Since this has been lying stale for 2 months I'm closing it, but feel free to reopen it if you end up working on it again.

**Pull request closed without merging**